src/ophyd_async/plan_stubs/_fly.py

```diff
@@ -126,18 +127,8 @@ def fly_and_collect(
 
     done = False
     while not done:
-        try:
-            yield from bps.wait(group=group, timeout=0.5)
-        except TimeoutError:
-            pass
-        else:
-            done = True
-        yield from bps.collect(
-            *detectors,
-            return_payload=False,
-            name=stream_name,
-        )
-    yield from bps.wait(group=group)
```

Comment:
The idea was to change this whole code block to `bps.collect_while_completing`, but I have since decided that it wasn't quite correct, so please can you copy https://github.com/bluesky/bluesky/blob/1b8422beee465d4f78f285a902b170a47303a75a/src/bluesky/plan_stubs.py#L1004-L1037 into ophyd-async, update to use it, and we will iterate from there